pantheon-bundle/src/test/java/com/redhat/pantheon/asciidoctor/extension/ModuleMetadataExtractorTreeProcessorTest.java

```diff
@@ -16,22 +16,22 @@
 import static org.junit.jupiter.api.Assertions.assertNull;
 
 @ExtendWith({SlingContextExtension.class, MockitoExtension.class})
-class MetadataExtractorTreeProcessorTest {
+class ModuleMetadataExtractorTreeProcessorTest {
 
     private final SlingContext slingContext = new SlingContext();
 
     @Test
     void extractMetadata() {
         // Given
         slingContext.build()
-                .resource("/content/module1/locales/en_US/1/metadata")
+                .resource("/content/module1/locales/en_US/1/moduleMetadata")
```

Comment:
hmmm... I don't like this. The name should not change to `moduleMetadata` as assemblies will also have this node.

Reply:
Good catch. Your comment made me think of another potential issue.

Let's say we are creating a new module from soup to nuts purely through the java model api. How does it know that some levels in the structure get named something meaningful - like the jcr node representing the module inheriting the file name - while other levels in the structure get named something static - like 'locale' or 'metadata'? I imagine that those static names need to appear in the java code somewhere - but where? As an annotation? As the name of the class itself? My concern is that it's not obvious, and perhaps renaming the class from "Metadata.class" to "ModuleMetadata.class" will accidentally rename the jcr nodes as well (Perhaps also true for ModuleLocale, etc...).

Thoughts there?

Reply:
To answer your first question re: creating nodes using the java api, the name of the node is dictated by the parent, the child node gets the name either from the variable name, or from the `@Named` annotation if one is present. This approach is static. The second approach is by creating methods on the interface which refer to the dynamically named nodes. You would then be able to get, create, remove these nodes via the API.

I don't mind renaming the class tbh, just not the node names themselves because then we are separating the two structures even more and at that point why even bother trying to keep them the same java classes for both.